modules/arm_plugin/src/arm_converter/arm_converter_one_hot.cpp

```diff
@@ -2,88 +2,111 @@
-            }
-        case ngraph::element::Type_t::u16 :
-            switch (outType) {
-                case ngraph::element::Type_t::u8  : return make(wrap_one_hot<std::uint16_t, std::uint8_t>);
-                case ngraph::element::Type_t::i16 : return make(wrap_one_hot<std::uint16_t, std::int16_t>);
-                case ngraph::element::Type_t::u16 : return make(wrap_one_hot<std::uint16_t, std::uint16_t>);
-                case ngraph::element::Type_t::i32 : return make(wrap_one_hot<std::uint16_t, std::int32_t>);
-                case ngraph::element::Type_t::f32 : return make(wrap_one_hot<std::uint16_t, float>);
-                default: THROW_IE_EXCEPTION << "Unsupported Type: " << outType; return {};
-            }
-        case ngraph::element::Type_t::u32 :
-            switch (outType) {
-                case ngraph::element::Type_t::u8  : return make(wrap_one_hot<std::uint32_t, std::uint8_t>);
-                case ngraph::element::Type_t::i16 : return make(wrap_one_hot<std::uint32_t, std::int16_t>);
-                case ngraph::element::Type_t::u16 : return make(wrap_one_hot<std::uint32_t, std::uint16_t>);
-                case ngraph::element::Type_t::i32 : return make(wrap_one_hot<std::uint32_t, std::int32_t>);
-                case ngraph::element::Type_t::f32 : return make(wrap_one_hot<std::uint32_t, float>);
-                default: THROW_IE_EXCEPTION << "Unsupported Type: " << outType; return {};
-            }
-        case ngraph::element::Type_t::i32 :
-            switch (outType) {
-                case ngraph::element::Type_t::u8  : return make(wrap_one_hot<std::int32_t, std::uint8_t>);
-                case ngraph::element::Type_t::i16 : return make(wrap_one_hot<std::int32_t, std::int16_t>);
-                case ngraph::element::Type_t::u16 : return make(wrap_one_hot<std::int32_t, std::uint16_t>);
-                case ngraph::element::Type_t::i32 : return make(wrap_one_hot<std::int32_t, std::int32_t>);
-                case ngraph::element::Type_t::f32 : return make(wrap_one_hot<std::int32_t, float>);
-                default: THROW_IE_EXCEPTION << "Unsupported Type: " << outType; return {};
-            }
-        default: THROW_IE_EXCEPTION << "Unsupported Type: " << node.get_input_element_type(0); return {};
+        const auto depth = depth_const -> cast_vector<int64_t>()[0];
+
+        if (ngraph::shape_size(ind_shape) * depth != ngraph::shape_size(out_shape))
+            THROW_IE_EXCEPTION << "Incompatible I/O shapes or wrong depth value.";
+        if (depth != out_shape[axis])
+            THROW_IE_EXCEPTION << "Incompatible depth and axis values.";
+
+        auto make = [&] (auto refFunction) {
+            return MakeConversion(refFunction,
+                                  node.input(0),
+                                  ind_shape,
+                                  node.output(0),
+                                  node.get_output_element_type(0).size(),
+                                  depth,
```

Comment:
```suggestion
                                  node.input(1),
```

Reply:
Thank you! Your suggestion was incorporated in the latest commit.